### PR TITLE
#36 - made requirements.txt the only one source of truth

### PR DESCRIPTION
**Pull request by eracle**

### CODE REVIEW

**Comment by eracle** [comment]
I will have a look on why it fails. I'll reply

**Comment by ydaniv** [comment]
@eracle it's failing because you rely on `requirements.txt` being present in the CI, and it isn't. Anyhow, this method looks like complicating things to me, I'd rather have it hard-coded like before. I'll send a review now.

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,6 @@ target/
 .idea/
 
 test.sh
+
+# ignoring virtualenvironment
```
Comment (ydaniv) [inline .gitignore]: This does not belong here, it's only relevant to your specific choice. Please remove.
As a side note, PEP8 requires an empty line at the end of the file.

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from setuptools import setup, find_packages
 
+with open('requirements.txt') as fp:
```
Comment (ydaniv) [inline setup.py]: No need for this here, the list is very simple and pretty much future proof for now, so I'd rather keep it simple as before.

**Comment by eracle** [comment]
I just have seen the Licence and also the way the owner manages PR, sorry I will not be part of this!

Ps. in the .gitignore file there is something which is also very specific to your choices:
`
.idea/
`

By the way if you want collaboration take into account the other's else need, in my case the need to ignore the `.venv` folder, which from my point of view, it didn't give you any disturbance.

**Pull request closed without merging**